### PR TITLE
Initial commit for the 3.4 release article.

### PR DESCRIPTION
**Pull request by paperdigits**
This is a placeholder for the 3.4 release article. We'll change the folder and article date when the release date is known. 

For the initial authoring, we're starting at https://hackmd.io/SEnOPyOeQ5i8xwXoPhYxYw to have as little friction as possible. Note that you must be logged in to hackmd to edit the article.

We'll eventually transition here to accept PRs on this article. 

Please do *NOT* use the github review comments to make comments about the article, it is a horrible workflow.

### CODE REVIEW

**Comment by ralfbrown** [comment]
Suggested first draft for the section tagging me:


* Various performance enhancements

** Many of the more compute-intensive image-processing algorithms have been updated to be faster and more scaleable when running on the CPU.  Everyone without OpenCL will see some improvement, and machines with eight or more cores will see major speedups.  The improved operations include nonlocal means denoising (used by both denoise(nonlocal means) and denoised(profiled) when set to nonlocal means), bilateral filter (used by local contrast, color mapping, global tonemapping, lowpass, monochrome, retouch, and shadows and highlights), and guided filter (used by haze removal and drawn mask feathering).

** Much unnecessary recomputation has been eliminated, leading to a more responsive user interface while editing an image in the darkroom.  The display of parametric and channel masks shows a particularly marked improvement.

**Comment by elstoc** [comment]
Thanks @ralfbrown I've added your suggestions to the WIP article on hackmd

**Comment by virtualritz** [comment]
http://www.hemingwayapp.com/ thinks this has readability 10 (highest score). Lower is better. I suggest at least cutting some of these endless sentences.
You can just copy & paste into the web version to get an idea.

File: content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+- The tethering view has been reworked to improve stability and now includes the histogram module again
+- A new button has been added to enable focus peaking mode, to complement the existing keyboard shortcut
+- Clicking on the darkroom history stack module's reset button provides a one-click way to discard history on your current image
+- The history stack can now be truncated -- removing all history items above the selected one but without compressing -- in the darkroom by holding _Ctrl_ while clicking the "compress history stack" button
+- New functionality has been added to `darktable-cli`
+- The filter categories combobox in the _collect images_ are now organized into groups
+- The data and library databases are now regularly checked for corruption
+- Backups of the data and library databases are now taken automatically allowing you to easily restore to an earlier state in case of corruption
+- By default the copy and paste buttons now exclude some modules that can cuase issues when pasted between images. You can still include these modules by using the "copy parts" and "paste parts" buttons
```
Comment (TurboGit) [inline content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md:176]: Only the copy excludes some module, I don't think we need to talk about paste here not "paste parts" at end of paragraph.

**Comment by TurboGit** [comment]
BTW, do you plan coordination with the other people making translation of the article each year. I know Nilvus was the main coordinator for the French article and there was also an Italian translation IIRC.

**Comment by paperdigits** [comment]
Yes, I had mentioned translation in my email to darktable-dev, but I will reach out to specific people in the next few days.

I hope the weekend time will allow anyone who has edits for the English to do so, then there will be less work for the translators.

**Comment by elstoc** [comment]
Might be nice to highlight that, if you include darktable, rawspeed and dtdocs, there have been more than 5500 commits in 2020.

File: content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+date: 2020-12-24
+lede: --
+lede_author: --
+tags: announcement, darktable-release
+
+Happy holidays everyone -- it's time for your favourite Christmast gift. This is the second major release of 2020 from the darktable project following the early release of darktable 3.2 in August, and we've been busy.
+
+# Documentation
+
+Photography can be obstuse and trying to manage your post-processing without documentation can make things even harder!  This time, though, the darktable team has been busy getting the user manual ready in time for the release and it is available today at https://www.darktable.org/resources/, and fully up-to-date with the latest version.
```
Comment (elstoc) [inline content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md:13]: Do you mean "abstruse" here?

File: content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md (hunk 1 above)
Comment (elstoc) [reply content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md:176]: Agreed - there's also a spelling mistake here: "cuase"

**Comment by elstoc** [comment]
And if anyone has a nice image we can use in the header...

**Comment by elstoc** [comment]
Could we add a link to @aurelienpierre's new video please

File: content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md (hunk 2 above)
Comment (paperdigits) [reply content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md:13]: I was going for "difficult to comprehend" which obtuse means, but it isn't the first definition.

**Comment by paperdigits** [comment]
@bilddateien @Nilvus it should be safe to begin translating this article, if you want to.

**Comment by paperdigits** [comment]
@elstoc link to YouTube video added.

File: content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md (hunk 2 above)
Comment (elstoc) [reply content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md:13]: I don't think that's the right definition. You can call a person obtuse meaning that they are slow.

**Comment by Nilvus** [comment]
àpaperdigits: you forgot @mpaglia0 for italian one...

For next article, it would be good to inform people before beginning the article and not when it's nearly over. Just notice that darktable.org website github is rarely followed by many of us as this repository rarely have new issues or PR...

Unfortunately, I will not have time to translate it this time.

**Comment by paperdigits** [comment]
Dull or without intellect is another meaning, yes. https://www.merriam-webster.com/dictionary/obtuse

But point taken we can change it.

**Comment by elstoc** [comment]
> For next article, it would be good to inform people before beginning the article and not when it's nearly over.

The 3.2 article PR was started about 3 weeks before 3.2 was released, which is similar in timescale to this one. The only difference is that this one has been publicly available and editable in draft form since the end of August.

**Comment by elstoc** [comment]
> @elstoc link to YouTube video added.

I assume this is in a local branch that you haven't pushed yet?

**Comment by Nilvus** [comment]
> > For next article, it would be good to inform people before beginning the article and not when it's nearly over.
> 
> The 3.2 article PR was started about 3 weeks before 3.2 was released, which is similar in timescale to this one. The only difference is that this one has been publicly available and editable in draft form since the end of August.

It was way before with french community, as it was done since few years (and I had just last year help to do that, I was not at the origin of that way to do). As I said this august, I'm totally ok that this is done that new way. Indeed, I don't care if it is done with french or english community. Just that we all have to learn and improve the way to do. That was just the reason for that. And that issue has been started at the end of august without communication for the first 3 months. Just a thing to improve for next article, no more.

**Comment by paperdigits** [comment]
@Nilvus an email was sent to darktable-dev on December 3rd saying this article was open for comment and that it was probably safe to start translating.

We have been discussing it in #darktable on IRC for a few months.

I'm not sure what other channels I should communicate in.

**Comment by Nilvus** [comment]
> @Nilvus an email was sent to darktable-dev on December 3rd saying this article was open for comment and that it was probably safe to start translating. We have been discussing it in #darktable on IRC for a few months. I'm not sure what other channels I should communicate in.

Thanks for your anwser. Just the same email that one you do in december at the beginning (so when you create that issue. Some of us, like me, never use IRC. The developer mailing list is probably the best way to, at least, launch the work. Nothing more.

**Comment by paperdigits** [comment]
It is probably best that people interested in things that are published to the website subscribe to the dtorg repo, that way you can be notified of what is happening with the website.

There is still plenty of time to suggest additions or changes. If you have them, please let them be known.

**Comment by paperdigits** [comment]
> I assume this is in a local branch that you haven't pushed yet?

Indeed you were correct. I've pushed it now.

File: content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md (hunk 2 above)
Comment (paperdigits) [reply content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md:13]: Changed.

File: content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md (hunk 1 above)
Comment (paperdigits) [reply content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md:176]: Fixed these two things.

**Comment by paperdigits** [comment]
> Might be nice to highlight that, if you include darktable, rawspeed and dtdocs, there have been more than 5500 commits in 2020.

I've mentioned this.

**Comment by bilddateien** [comment]
> @bilddateien @Nilvus it should be safe to begin translating this article, if you want to.

Thanks for the hint.
But I'm wondering about where to start. I find https://hackmd.io/SEnOPyOeQ5i8xwXoPhYxYw to not be in sync with e. g. https://github.com/darktable-org/dtorg/edit/3dot4-release-article/content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md 

On a sidenote I'm also thinking about how to deal with this text since it uses links to the new doc extensively. But this one is available in English only at this point afaik.
Perhaps I could add some extra explainatory notes into the translated text ...

**Comment by paperdigits** [comment]
You should use the version in git; we are no longer editing the hackmd version.

**Comment by bilddateien** [comment]
> You should use the version in git; we are no longer editing the hackmd version.

ok - I assumed that.
I put a preliminary translation here https://www.bilddateien.de/blog/2020-12-24-darktable-3-4-weiteres-update-preliminary.html and added some screenshots.

I could provide those in english and french also if needed. Pls. let me know.

**Comment by bilddateien** [comment]
Just corrected the user-manual links.
German translation is ready to go, final link will be https://www.bilddateien.de/blog/2020-12-24-darktable-3-4-weiteres-update.html 4 days from now ...

**Comment by wpferguson** [comment]
luadocs are in place including the lua API manual, thanks to the nudge from @TurboGit (removing the user manual which had the lua API manual embedded) :smile:

File: content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+- Altered preferences (those that have been changed from their default values) are now indicated with a bullet symbol
+- If preferences are changed that require a restart in order to take effect, a message will appear to remind you to restart when exiting the preferences dialog
+- A number of preferences have been altered such that they no longer require a restart to take effect
+- The 'CSS theme tweaks' functionality in the General tab now has a more intuitive workflow -- press the "save and apply" button to apply your CSS immediately without having to first select the "modify theme with CSS tweaks" checkbox. This checkbox can still be used to temporarily disable your modifications in case of error.
+
+# Presets
+
+**needs more words and discussion of other preset changes**
+
```
Comment (elstoc) [inline content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md:146]: Please delete this line

**Comment by paperdigits** [comment]
> German translation is ready to go, final link will be https://www.bilddateien.de/blog/2020-12-24-darktable-3-4-weiteres-update.html 4 days from now ...

Just added a link to it.

File: content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md (hunk 3 above)
Comment (paperdigits) [reply content/blog/2020-12-24-darktable-3.4/2020-12-24-darktable-3.4.md:146]: Done!

**Pull request merged**